kii_thing_if.c

```diff
@@ -277,6 +278,30 @@ static prv_bool_t prv_set_firmware_version_resource_path(
   return TRUE;
 }
 
+static prv_bool_t prv_set_thing_type_resource_path(
+        kii_t* kii,
+        char* resource_path,
+        size_t resource_path_len)
+{
+    if (resource_path_len <=
+            CONST_STRLEN(THING_IF_APP_PATH) +
```

Comment:
Why do you put those procedure inside the condition statement?
(Seems you prefer to do so in general.)

I am not a strong supporter of this style because I think this makes hard to read, modify and debug the code.